### PR TITLE
Client support for targeted signals

### PR DESCRIPTION
**Pull request by WillieHabi**
## Description

Client side changes needed to support targeting signals to a specific client id. 

These changes follow the server changes to support targeted signals https://github.com/microsoft/FluidFramework/pull/19519

### CODE REVIEW

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 1]
```diff
@@ -345,7 +346,12 @@ export class DocumentDeltaConnection
 			throw new UsageError("Sending signals to specific client ids is not supported.");
 		}
 
-		this.emitMessages("submitSignal", [[content]]);
+		const signal: ISentSignalMessage = {
+			content,
+			targetClientId,
+		};
+
+		this.emitMessages("submitSignal", [signal]);
```
Comment (jason-ha) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts]: I don't think we can blanket submit `ISentSignalMessage[]`. The shape of the received content won't be correct if the service doesn't support v2.
The targetClientId and v2 check above is incorrect per chat https://teams.microsoft.com/l/message/19:0f94bd601565474bab29154a77a4528f@thread.v2/1712080563331?context=%7B%22contextType%22%3A%22chat%22%7D -- more visible discussion at https://github.com/microsoft/FluidFramework/pull/17729/files#r1533281026

File: packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts [hunk 2]
```diff
@@ -191,3 +196,224 @@ describeCompat("TestSignals", "FullCompat", (getTestObjectProvider) => {
 		);
 	});
 });
+
+describeCompat.skip("TargetedSignals", "NoCompat", (getTestObjectProvider) => {
```
Comment (jason-ha) [inline packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts:203]: Relocating comment from #19367:
> Everything is skipped. Is there nothing that can be tested? (Hint we definitely should be able to test current behavior.)

I am not sure why #19367 was closed. There is always value in setting up tests before making functional changes. Perhaps things are unblocked if that targetClientId and feature check were removed from DocumentDeltaConnection - we think check is bogus.

Comment (jason-ha) [reply packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts:203]: Or perhaps those tests are not quite the right ones for existing state of the world.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts (hunk 2 above)
Comment (WillieHabi) [reply packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts:203]: Right, I will re-open and update it!

**Pull request closed without merging**